### ISSUE TITLE
Include documentation for `assertComplete` and `assertFailure`  methods 

### ISSUE THREAD

**Issue by ThorodanBrom** [eclipse-vertx/vertx-junit5#140]
#### Describe the feature

- Since Vert.x 5.0 is deprecating the callback model, I was wondering **if the callbacks used in methods like `succeeding` and `failing` would stop working in 5.0** (or maybe I have not understood the deprecation properly)
- I found out about the existence of the methods `assertComplete` and `assertFailed` developed by @slinkydeveloper from their [blog](https://slinkydeveloper.com/Assertions-With-Vertx-Futures-And-JUnit5/), which allow for testing with the help of Futures instead of the callbacks
- However, there isn't any official documentation about these methods in the official docs

#### Use cases

- Including documentation for this may help those who are looking to move away from `Handler<AsyncResult>` -style callbacks in JUnit tests
- If `succeeding`, `failing` are getting deprecated, then `assertComplete` and `assertFailing` would be the only methods usable (I think I am wrong here)

#### Contribution

- I can try to add documentation based on @slinkydeveloper's blog post

**Comment by tsegismont** [comment]
No, these methods won't stop working, it's perfectly fine to use them, for example, at the the end of a future composition chain.

Note that in Vert.x 5 you can create Future expectations with [`Future.expecting`](https://vertx.io/docs/5.0.0.CR2/apidocs/io/vertx/core/Future.html#expecting(io.vertx.core.Expectation))

It would be great if you could contribute a doc update about test context methods related to futures.
